backend/src/main/java/com/github/redstylzz/backend/service/PaymentService.java

```diff
@@ -30,44 +34,60 @@ private boolean paymentExists(String paymentID) {
         return paymentRepo.existsByPaymentID(paymentID);
     }
 
+    private void calculatePaymentSum(String userID, String categoryID) {
+        List<Payment> payments = paymentRepo.getAllByUserIDAndCategoryID(userID, categoryID);
+        BigDecimal sum = payments.stream().map(Payment::getAmount).reduce(BigDecimal.ZERO, BigDecimal::add);
+        categoryService.setCategorySum(userID, categoryID, sum);
+    }
+
+    private List<PaymentDTO> getPaymentAsDTO(String userID, String categoryID) {
```

Comment:
this method name is a bit bulky and does clearly state your intention, you may want to consider getPaymentDTOForUserIdAndcategoryId